sklearn/utils/_array_api.py

```diff
@@ -232,6 +271,18 @@ def take(self, X, indices, *, axis=0):
     def isdtype(self, dtype, kind):
         return isdtype(dtype, kind, xp=self._namespace)
 
+    def searchsorted(self, a, v, *, side="left", sorter=None):
+        # Temporary workaround needed as long as searchsorted is not part
+        # of the Array API spec:
+        # https://github.com/data-apis/array-api/issues/688
```

Comment:
```suggestion
        # Temporary workaround needed as long as searchsorted is widely adopted
        # by implementers of the Array API spec.
        # This is a quite recent addition to the spec:
        # https://data-apis.org/array-api/latest/API_specification/generated/array_api.searchsorted.html
```

But we actually need to check if this fallback is actually needed at all. If a compliant implementation of `searchsorted` is available in all the libraries we currently test against, I would vote for removing this generic fallback implementation entirely.

Reply:
Hi @ogrisel 

From your comments it seems like we can ignore array-api-strict and we don't need to test for that. However even if we ignore that, I was just checking cupy.array_api functions and I could not find any of the functions that I have defined in this PR in those https://docs.cupy.dev/en/stable/reference/array_api_functions.html, including searchsorted. If we can ignore cupy array api as well then we might be able to remove these custom implementations.

Reply:
We need cupy support, so we can't skip it.

I think the comment in Olivier's suggestion should be "Temporary workaround needed as long as searchsorted is not widely adopted" - the change being an added "not".